src/cogent3/core/alignment.py

```diff
@@ -2285,6 +2298,103 @@ def get_features(
                 # passing self only used when self is an Alignment
                 yield seq.make_feature(feature, self)
 
+    def distance_matrix(self, calc="jaccard", k: int = 7):
```

Comment:
We should not include `k` as an argument since the coefficients are derived from a specific value of `k`. The docstring needs to declare this. We have to hard-code the value of k that was used to obtain the coefficients.

`calc` is either `pdist` or `jc69` where the doctsring makes it clear the latter is a transformation of the former.
We should also raise a `TypeError` if moltype is not either DNA or RNA.